src/BinaryOperator.php

```diff
@@ -76,4 +76,34 @@ public function toQuery(): string
      * @return string
      */
     abstract protected function getOperator(): string;
+
+    /**
+     * Gets the left-hand of the expression.
+     *
+     * @return AnyType
+     */
+    public function getLeft(): AnyType
+    {
+        return $this->left;
+    }
+
+    /**
+     * Gets the right-hand of the expression.
+     *
+     * @return AnyType
+     */
+    public function getRight(): AnyType
+    {
+        return $this->right;
+    }
+
+    /**
+     * Returns whether or not the operator inserts parenthesis.
+     *
+     * @return bool
+     */
+    public function insertsParentheses(): bool
```

Comment:
I think `insertsParentheses` is a bit misleading. Consider renaming it to `hasParentheses`.